plugins/balloontoolbar/plugin.js

```diff
@@ -528,6 +545,17 @@
 			this._contexts.splice( 0, this._contexts.length );
 		},
 
+		/**
+		 * Refreshes all contexts from {@link #_contexts}.
+		 *
+		 * @private
+		 */
+		_refresh: function() {
```

Comment:
All other `refresh` methods are public, only that one is private. What was the motivation behind such decision?

Reply:
Because this one is only used internally by `ContextManager`. `ContextManager` is the one managing `balloontoolbar` deciding when to show/hide it so it also manages commands state related to `balloontoolbar` controls. That's why I thought it should be `ContextManager` private method.

Reply:
Makes sense.